krabs/krabs/filtering/comparers.hpp

```diff
@@ -64,10 +60,20 @@ namespace krabs { namespace predicates {
             template <typename Iter1, typename Iter2>
             bool operator()(Iter1 begin1, Iter1 end1, Iter2 begin2, Iter2 end2) const
             {
-                auto r1 = boost::make_iterator_range(begin1, end1);
-                auto r2 = boost::make_iterator_range(begin2, end2);
+                const auto dist1 = std::distance(begin1, end1);
```

Comment:
This works for me here:
```C++
const auto dist1 = std::distance(begin1, end1);
const auto dist2 = std::distance(begin2, end2);
return dist1 >= dist2 && std::equal(begin1, begin1 + dist2, begin2, Comparer());
```
Note that std::equal() returns true when distance is 0.

Reply:
Take a look at std::mismatch - it may be possible to use that instead of having to move the iterators around.
I wouldn't rely on iterator math working - that only works with random access iterators.

Reply:
I appreciate the std::mismatch suggestion - rewrote it in terms of that algorithm and it is much more concise.